modules/plugins/utility/ccc/ccc.nix

```
@@ -1,10 +1,154 @@
 {lib, ...}: let
-  inherit (lib.options) mkEnableOption;
+  inherit (lib.options) mkOption mkEnableOption;
+  inherit (lib.types) listOf enum;
   inherit (lib.nvim.binds) mkMappingOption;
+  inherit (lib.nvim.types) mkPluginSetupOption luaInline;
+  inherit (lib.generators) mkLuaInline;
 in {
   options.vim.utility.ccc = {
     enable = mkEnableOption "ccc color picker for neovim";
 
+    setupOpts = mkPluginSetupOption "ccc.nvim" {
+      highlighter = mkOption {
+        type = luaInline;
+        default = mkLuaInline ''
+          {
+            auto_enable = true,
+            max_byte = 2 * 1024 * 1024, -- 2mb
+            lsp = true,
+            filetypes = colorPickerFts,
+          }
+        '';
+        description = "Settings for the highlighter";
+      };
+
+      pickers = mkOption {
+        type = luaInline;
+        default = mkLuaInline ''
+          {
+            ccc.picker.hex,
+            ccc.picker.css_rgb,
+            ccc.picker.css_hsl,
+            ccc.picker.ansi_escape {
+              meaning1 = "bright", -- whether the 1 means bright or yellow
+            },
+          }
+        '';
+        description = ''
+          List of formats that can be detected by |:CccPick| to be activated.
```

```suggestion
          List of formats that can be detected by {command}`:CccPick` to be activated.
```

NDG doesn't have a way of processing Neovim's help text yet, so it's nicer to parse this as a command role. In the future I might add a postprocessor that links the appropriate Neovim documentation from `|:cmd|`.